apps/bemo-worker/src/BemoDO.ts

```diff
@@ -13,16 +14,31 @@ const connectRequestQuery = T.object({
 	storeId: T.string.optional(),
 })
 
+interface AnalyticsEvent {
+	type: 'connect' | 'send_message' | 'receive_message'
+	origin: string
+	sessionKey: string
+}
+
 export class BemoDO extends DurableObject<Environment> {
 	r2: R2Bucket
 	_slug: string | null = null
 
+	analytics: AnalyticsEngineDataset
+
+	writeEvent({ type, origin, sessionKey }: AnalyticsEvent) {
+		this.analytics.writeDataPoint({
+			blobs: [type, origin, sessionKey],
```

Comment:
what's the value in writing the session key? I could maybe see room ID, but i'm not sure where sessionkey would be actionable

Reply:
i was thinking it would allow us to see how many different clients there independently of how many connect events there are. but also yes I added the room id, silly me for forgetting it.